meson.build

```diff
@@ -301,20 +301,21 @@ endforeach
 
 if cc.has_function('strerror_r', prefix: '#include <string.h>')
   strerror_r_code = '''
+#define _GNU_SOURCE
```

Comment:
Is this necessary, given that we supply `-D_GNU_SOURCE` through `add_project_arguments`?

Reply:
Yes - note in [the documentation for add_project_arguments](https://mesonbuild.com/Adding-arguments.html), it says <q>Compilation tests that are run as part of your project configuration do not use these flags.</q>

Using current master on my machine with GCC 13.1 and glibc 2.37, Meson builds this test with ```cc .../testfile.c -o /home/ats/src/ .../output.exe -D_FILE_OFFSET_BITS=64 -O0```, so it detects the XSI version, but then goes on to build the project itself with `-D_GNU_SOURCE`, so the code that uses `strerror_r` will be wrong if built with Meson as it stands.

Reply:
I see, thank you.